src/github/types.ts

```diff
@@ -2,20 +2,32 @@ export type Repository = { owner: string; repo: string };
 
 export type Issue = { number: number; node_id: string };
 
+/** Key value pair with the name/id of a field and the name/id of its value */
+export type FieldValues = { field: string; value: string };
+
+export type NodeData = { id: string; title: string };
+
 export interface IProjectApi {
+  /**
+   * Fetches the node id from the project id and caches it.
+   * @returns node_id of the project. This value never changes so caching it per instance is effective
+   */
+  fetchProjectData(): Promise<NodeData>;
   /**
    * Assign an issue to a project
    * @param issue The issue object which has the number and the node_id
    */
-  assignIssue(issue: Issue): Promise<boolean>;
-
+  assignIssue(issue: Issue, project: NodeData): Promise<string>;
```

Comment:
tbh, not really clear, what is the `string`, that's returned here.  
How about adding `@returns`, or maybe a type alias for what-is-returned there